web-api/src/persistence/dynamo/users/createOrUpdateUser.ts

```diff
@@ -117,14 +115,12 @@ export const isUserAlreadyCreated = async ({
   email,
   userPoolId,
 }: {
-  applicationContext: IApplicationContext;
+  applicationContext: ServerApplicationContext;
   email: string;
   userPoolId: string;
 }) => {
-  const cognito: CognitoIdentityProvider = applicationContext.getCognito();
-
   try {
-    await cognito.adminGetUser({
+    await applicationContext.getCognito().adminGetUser({
       UserPoolId: userPoolId,
       Username: email,
     });
```

Comment:
should this be behind the user gateway? if that `getUser()` function returns nothing, then it can be assumed it's not taken